lib/active_model/serializer/collection_serializer.rb

```diff
@@ -44,8 +44,16 @@ def json_key
           explicit_serializer_class._type
         # 2. get from first serializer instance in collection
         key ||= (serializer = serializers.first) && serializer.json_key
-        # 3. get from collection name, if a named collection
-        key ||= object.respond_to?(:name) ? object.name && object.name.underscore : nil
+        if object.respond_to?(:name)
```

Comment:
Thanks, was a bit messy as I wrote it... what do you think about changing to throw/catch?